ambry-server/src/main/java/com.github.ambry.server/AmbryServer.java

```diff
@@ -190,6 +197,9 @@ public void shutdown() {
           logger.error("Error while closing notification system.", e);
         }
       }
+      if (statsManager != null) {
```

Comment:
shut this down first. General rule of thumb to follow - whatever starts last shuts down first. Otherwise, stores may shutdown and stats manager can still be calling `getStatsSnapshot()`